aptos-move/framework/aptos-framework/sources/randomness.move

```diff
@@ -0,0 +1,23 @@
+/// On-chain randomness utils.
+module aptos_std::randomness {
+    use aptos_framework::system_addresses;
+
+    friend aptos_framework::block;
+
+    /// The block-level seed randomness.
+    /// It's updated at the beginning of every block.
+    struct BlockRandomness has drop, key {
+        block_randomness: vector<u8>,
+    }
+
+    /// Invoked in `block_prologue_ext()` to update the block-level seed randomness.
+    public(friend) fun on_new_block(account: &signer, randomness_available: bool, block_randomness: vector<u8>) acquires BlockRandomness {
+        system_addresses::assert_aptos_framework(account);
+        if (exists<BlockRandomness>(@aptos_framework)) {
+            move_from<BlockRandomness>(@aptos_framework);
+        };
+        if (randomness_available) {
```

Comment:
In which scenarios can this be false?

Reply:
this is false when at the start of a block we don't have randomness,
which is when
- either (1) randomness generation feature is disabled (i think it should be a separate feature flag )
- or (2) we entered the current epoch without DKG results

(2) happens when:
- at any time, a txn executes `reconfiguration::reconfigure()` directly.
  - This is probably a bug: it's should probably call `reconfiguration_with_dkg::start()`
- DKG times out (but we can configure it to never time out)